packages/x-tree-view/src/internals/plugins/useTreeViewNodes/useTreeViewNodes.ts

```diff
@@ -138,12 +148,16 @@ export const useTreeViewNodes: TreeViewPlugin<UseTreeViewNodesSignature> = ({
     [instance],
   );
 
-  const getChildrenIds = useEventCallback((nodeId: string | null) =>
-    Object.values(state.nodes.nodeMap)
-      .filter((node) => node.parentId === nodeId)
-      .sort((a, b) => a.index - b.index)
-      .map((child) => child.id),
-  );
+  const getChildrenIds = useEventCallback((nodeId: string | null) => {
```

Comment:
Nice side effect of storing the indexes in this new structure, we no longer have to loop on **all** the items to find the one that belong to a certain parent.